### PR TITLE
Add 'DidEndPreview' listener to utils.is.openFile

### PR DESCRIPTION
**Pull request by jzgoda**
Implements #3492.

### CODE REVIEW

File: tns-core-modules/utils/utils.ios.ts [hunk 1]
```diff
@@ -185,6 +192,7 @@ export function openUrl(location: string): boolean {
 
 class UIDocumentInteractionControllerDelegateImpl extends NSObject implements UIDocumentInteractionControllerDelegate {
     public static ObjCProtocols = [UIDocumentInteractionControllerDelegate];
+    public static didEndPreview = new Observable();
```
Comment (hshristov) [inline tns-core-modules/utils/utils.ios.ts]: Could you try moving this `Observable` as instance field and attach it to the `controller`?
Then in `documentInteractionControllerDidEndPreview` you could take it from the `controller` parameter.

Comment (jzgoda) [inline tns-core-modules/utils/utils.ios.ts]: Can you provide me with some additional guidance as to what you mean by attaching the variable to the `controller`? (I'm not familiar with `UIDocumentInteractionController` or Obj-C.)

But, I can turn it into an instance variable like this:
```javascript
export function openFile(filePath: string, callback?: Function): boolean {
    ...

        var controller = UIDocumentInteractionController.interactionControllerWithURL(NSURL.fileURLWithPath(path));
        var delegate = new UIDocumentInteractionControllerDelegateImpl();
        controller.delegate = delegate;
        if (callback) {
            delegate.didEndPreview.on('didEndPreviewEventName', () => {
                delegate.didEndPreview.off('didEndPreviewEventName');
                callback();
            });
        }  

    ...
}
```
```javascript
class UIDocumentInteractionControllerDelegateImpl extends NSObject implements UIDocumentInteractionControllerDelegate {
    public static ObjCProtocols = [UIDocumentInteractionControllerDelegate];
    public didEndPreview = new Observable();

    ...

    public documentInteractionControllerDidEndPreview(controller: UIDocumentInteractionController) {
        let eventData: EventData = {
            eventName: "didEndPreviewEventName",
            object: this.didEndPreview
        };
        this.didEndPreview.notify(eventData);
    }
}
```

Would that be better?

Comment (jzgoda) [inline tns-core-modules/utils/utils.ios.ts]: @hshristov, I tried these changes that I just presented, and they don't seem to work...the `delegate` doesn't seem to be getting created properly. 

However, my original PR does work, and I'm still not completely sure what alternative route you want me to go down. Please let me know how I can help to get this accepted.  
Thanks.

**Comment by vakrilov** [comment]
Closing due to inactivity and really really hard rebase.

@jzgoda If you are willing to give this one another try and base your changes on the latest code in master - it would be a blast.

You can also check our new [contributing guide](https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md)

**Comment by jzgoda** [comment]
@vakrilov I'll be doing some work on the app that uses this feature in the next couple of months, so I will give it another shot with the latest version.

**Pull request closed without merging**